packages/core/src/Modules/Cashier/Components/on-ramp-provider-card.jsx

```diff
@@ -1,43 +1,98 @@
+                            custom_color = provider.dark_mode_fill;
+                        } else {
+                            custom_color = provider.light_mode_fill;
+                        }
+                    }
+                    return <Icon key={idx} size={40} icon={provider.icon} custom_color={custom_color} />;
+                })}
             </div>
         )}
-        <Button
-            id={`gtm-onramp-provider-select--${provider.name.toLowerCase().replace(' ', '-')}`}
-            className='on-ramp__provider-button'
-            type='button'
-            has_effect
-            onClick={() => setSelectedProvider(provider)}
-            text={localize('Select')}
-            primary
-        />
+    </div>
+);
+
+const OnRampProviderCard = ({ provider, setSelectedProvider, is_dark_mode_on }) => (
+    <div className='on-ramp__provider'>
+        <DesktopWrapper>
+            <div className='on-ramp__provider-logo'>
+                <Icon icon={provider.icon} width={118} height={28} />
+            </div>
+            <h2 className='on-ramp__provider-name'>{provider.name}</h2>
+            <div className='on-ramp__provider-description'>{provider.description}</div>
+            <a
+                href={provider.website}
+                className='on-ramp__provider-website link'
+                target='_blank'
+                rel='noopener noreferrer'
+            >
+                {provider.website}
+            </a>
+            <ProviderIcons providers={provider.payment_icons} is_dark_mode_on={is_dark_mode_on} />
+            <Button
+                id={`gtm-onramp-provider-select--${provider.name.toLowerCase().replace(' ', '-')}`}
+                className='on-ramp__provider-button'
+                type='button'
+                has_effect
+                onClick={() => setSelectedProvider(provider)}
+                text={localize('Select')}
+                primary
+            />
+        </DesktopWrapper>
+        <MobileWrapper>
+            <div className='on-ramp__provider-content'>
+                <div className='on-ramp__provider-logo'>
+                    <Icon icon={provider.icon} width={56} height={56} />
+                </div>
+                <div className='on-ramp__provider-detail'>
+                    <h2 className='on-ramp__provider-name'>{provider.name}</h2>
+                    <div className='on-ramp__provider-description'>{provider.description}</div>
+                    <a
+                        href={provider.website}
+                        className='on-ramp__provider-website link'
+                        target='_blank'
+                        rel='noopener noreferrer'
+                    >
+                        {provider.website}
+                    </a>
+                </div>
+            </div>
+
+            <div className='on-ramp__provider-footer'>
+                <ProviderIcons providers={provider.payment_icons} is_dark_mode_on={is_dark_mode_on} />
```

Comment:
Probably should rename `providers` prop to icons as it's expecting icons rather than providers.